app/Services/DeactivateAccountService.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Models\User;
+use App\Repositories\UserRepository;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class DeactivateAccountService
+{
+    protected $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function execute(Request $request, User $user)
```

Comment:
I think we don't need to send the user id in the request as it is already action allowed only by the logged in user and if the user is authenticated, just deactivate the user account end expire the token. What do you think?